### PR TITLE
[Clang] Don't assert non-empty packs for FunctionParmPackExprs

### LINKED ISSUE #86361: Assertion failure with nested parameter packs and lambdas
Another similar to https://github.com/llvm/llvm-project/issues/61460 (in fact the original internal reproduce tripped over both #61460 and this new bug - in that once #61460 was fixed, the original reproducer still crashed, so I had to reduce a new reproducer for this bug)
```
template <typename... Ts>
void f1(Ts... ts) {
  [&](auto... indexes) {
    ([&] {
        (ts);
        indexes;
      }, ...);
  };
}
void f2() { f1(); }
```
https://godbolt.org/z/6KW6jT1n7

### PR DESCRIPTION
**Pull request by zyn0217**
`FunctionParmPackExpr`s are peculiar in that they have to be of unexpanded dependency while they don't introduce any unexpanded packs. So this patch rules them out in the non-empty pack assertion in `DiagnoseUnexpandedParameterPack()`

There was a fix #69224, but that turned out to be insufficient.

I also moved the separate tests to a pre-existing file.

Fixes https://github.com/llvm/llvm-project/issues/86361

### CODE REVIEW

File: clang/test/SemaCXX/pr61460.cpp [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-template <typename... Ts> void g(Ts... p1s) {
-  (void)[&](auto... p2s) { ([&] { p1s; p2s; }, ...); };
-}
-
-void f1() {
-  g();
-}
-
-template <typename... Ts> void g2(Ts... p1s) {
-  (void)[&](auto... p2s) { [&] { p1s; p2s; }; }; // expected-error {{expression contains unexpanded parameter pack 'p2s'}}
```
Comment (erichkeane) [inline clang/test/SemaCXX/pr61460.cpp:12]: I see this test isn't above, is it?  Why doesn't this contain an unexpanded parameter pack anymore?  OR, should this test be added above too?

**Review by erichkeane — COMMENTED** [review]
This looks reasonable, I want to think on it a little and give others a chance to look at it, plus I want ot make sure we didn't lose test coverage.  Else LGTM.

**Review by erichkeane — APPROVED** [review]
I think with the test, I'm happy.  Please give other folks ~24 hrs to get a chance to take a look in case there is something I missed.

**Comment by zyn0217** [comment]
It's been ~3 days and I'm merging it - feel free to do post-commit reviews.

**Pull request merged**